### ISSUE TITLE
The seg-pretraining issues

### ISSUE THREAD

**Issue by Key-lei** [AImageLab-zip/alveolar_canal#8]
Hello author! I have an error in code reproduction that I would like to get your answer to. When I run the command `python main.py --configs config/seg-pretraining.yam`l, I get a divide by zero error, the error comes from `expierments `line 204 running continue operation on the whole dataset, what causes this? Because `torch.sum(gt_count) `is always equal to 0, and does synthetic_loader only use dense tags generated by gen-inference.yaml? Looking for your answer to the above error when trying to reproduce your work.

**Comment by LucaLumetti** [comment]
Hi @Key-lei,
looks like [line 204 of experiments.py](https://github.com/AImageLab-zip/alveolar_canal/blob/190af1c7f17aa5625eccce722e2104497b2f7e69/experiments/experiment.py#L204) is empty. I would assume the error is thrown by [line 216 of the same file](https://github.com/AImageLab-zip/alveolar_canal/blob/190af1c7f17aa5625eccce722e2104497b2f7e69/experiments/experiment.py#L216).
If all the data inside the generated dataset is empty, it probably means that something during the Deep Label Expansion (or Generation) phase has gone wrong. If the inference has been performed using checkpoints calculated by you, please check the metrics you obtained in this phase and check that the network didn't overfit on predicting always empty volumes. As I've noticed that you have added/deleted some lines of code, be sure you didn't break anything.

synthetic_loader should be a set of SubjectImage containing the CBCT data of volume contained in the synthetic split together with their generated dense label (as they do not have dense annotations produced by medical experts).

**Comment by Key-lei** [comment]
Thank you for your reply! Also, I noticed that after executing `gen-training.yaml` and executing `gen-inference.yaml `some of the patients under the sparse file generated` generated.npy files` but still most of the patients did not have `generated.npy `is this the cause of the error? I kept all the code the same as you provided and executed `gen-training.yaml` and `gen-inference.yaml` in order and it worked, but when I executed `seg-pretraining.yaml` I got an error.

**Comment by Key-lei** [comment]
Thank you for your answer! Currently I changed the `split` in the dataset of the `inference function `under the `generation.py` file in the `experiments` folder from `splits=['train', 'val', 'test']` to `splits='synthetic',` ran gen-inference I don't know if this principle is causing my problem, but when I run the splits under the original inference, only 91 dense annotations of synthetic are generated. Thanks again for your answer, and I hope to hear from you again if I have any follow-up questions. have a nice life!

**Comment by puppy2000** [comment]
> Thank you for your answer! Currently I changed the `split` in the dataset of the `inference function `under the `generation.py` file in the `experiments` folder from `splits=['train', 'val', 'test']` to `splits='synthetic',` ran gen-inference I don't know if this principle is causing my problem, but when I run the splits under the original inference, only 91 dense annotations of synthetic are generated. Thanks again for your answer, and I hope to hear from you again if I have any follow-up questions. have a nice life!

Are you still doing well in the subsequent training process?
